apps/rush-lib/src/api/PackageManagerFeatureSet.ts

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as semver from 'semver';
+import { RushConstants } from '../logic/RushConstants';
+
+/**
+ * This represents the available Package Manager tools as a string
+ * @public
+ */
+export type PackageManager = 'pnpm' | 'npm' | 'yarn';
+
+/**
+ * Reports the known features of a package manager as detected from its version number.
+ * @beta
+ */
+export class PackageManagerFeatureSet {
+  /**
+   * The package manager.
+   */
+  public readonly packageManager: PackageManager;
+
+  /**
+   * The SemVer version of the package manager.
+   */
+  public readonly version: string;
+
+  /**
+   * The filename of the shrinkwrap file that is used by the package manager.
+   *
+   * @remarks
+   * Example: `npm-shrinkwrap.json` or `pnpm-lock.yaml`
+   */
+  public readonly shrinkwrapFilename: string;
+
+  /**
+   * PNPM only.  True if `--resolution-strategy` is supported.
+   */
+  public readonly supportsPnpmResolutionStrategy: boolean;
+
+  public constructor(packageManager: PackageManager, version: string) {
+    this.packageManager = packageManager;
+    this.version = version;
+
+    const parsedVersion: semver.SemVer = new semver.SemVer(version);
+
+    this.supportsPnpmResolutionStrategy = false;
+
+    switch (this.packageManager) {
```

Comment:
It kinda feels like this data should be expressed in JSON files. TS for code, JSON for data.

Reply:
Maybe, but let's collect a little more "data" before we design a specialized manager for it

Reply:
Sure - that's fine.